### PR TITLE
Fix deploying machines

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3309. Fixes: #3308.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- start a new model
- open the machine view and add a machine
- deploy
- the model should deploy correctly

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8247/

File: jujugui/static/gui/src/app/components/generic-input/generic-input.js [hunk 1]
```diff
@@ -212,7 +212,7 @@ class GenericInput extends React.Component {
     }
     return (
       <input className="generic-input__field"
-        autoComplete={this.props.autocomplete}
+        autoComplete={this.props.autocomplete ? 'on' : 'off'}
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/generic-input/generic-input.js]: Thanks for fixing this

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 2]
```diff
@@ -489,43 +489,46 @@ Browser: ${navigator.userAgent}`
     }
 
     const entityId = state.postDeploymentPanel.entityId;
-    const nowMillis = new Date().getTime();
+    // If there are no new entites then there's no need to display the panel.
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: entities

Comment (hatched) [reply jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Why wouldn't there be an entity? Can you add a comment to this effect, or maybe it's another bug?

Comment (huwshimi) [reply jujugui/static/gui/src/app/init/component-renderers-mixin.js]: I presume there are no entities any time you're not deploying new applications. At the least in this case my model only contained a machine.

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8252/

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**